pr-deploy/test/app.test.ts

```diff
@@ -17,115 +17,145 @@
 // mock unzipAndMove dependency before importing '../src/app'
 jest.mock('../src/zipper', () => {
   return {
-    unzipAndMove: jest.fn().mockReturnValue(Promise.resolve('gs://serving-bucket/travisBuildNumber')),
+    unzipAndMove: jest
+      .fn()
+      .mockReturnValue(
+        Promise.resolve('gs://serving-bucket/travisBuildNumber')
+      ),
   };
 });
 
-import {Application} from 'probot';
+process.env.GH_CHECK = 'test-check';
+process.env.GH_OWNER = 'test-owner';
+process.env.GH_REPO = 'test-repo';
+
+import nock from 'nock';
 import prDeployAppFn from '../src/app';
-import express from 'express';
-import request from 'supertest';
-import Webhooks, {
-  WebhookPayloadPullRequest,
-  WebhookPayloadCheckRun} from '@octokit/webhooks';
+import Webhooks from '@octokit/webhooks';
+import {Probot, ProbotOctokit} from 'probot';
+
+const apiUrl = 'https://api.github.com';
 
 describe('test pr deploy app', () => {
-  let app: Application;
-  let github: any;
-  let server: express.Application;
+  let probot: Probot;
 
   beforeEach(() => {
-    process.env.DISABLE_WEBHOOK_EVENT_CHECK = 'true';
-
-    github = {
-      checks: {
-        create: jest.fn().mockReturnValue(Promise.resolve()),
-        update: jest.fn().mockReturnValue(Promise.resolve()),
-        listForRef: jest.fn(),
-      },
-    };
-
-    app = new Application();
-    app.load(prDeployAppFn);
-    app.auth = async() => Promise.resolve(github);
+    nock.disableNetConnect();
+    probot = new Probot({
+      id: 1,
+      githubToken: 'test',
+      // Disable throttling & retrying requests for easier testing
+      Octokit: ProbotOctokit.defaults({
+        retry: {enabled: false},
+        throttle: {enabled: false},
+      }),
+    });
+    probot.load(prDeployAppFn);
+  });
 
-    server = express();
-    server.use(app.router);
+  afterEach(() => {
+    nock.cleanAll();
+    nock.enableNetConnect();
```

Comment:
I've encountered issues in the past where a nocked endpoint is never called (including the reply handler that has `expects` on the response), and as a result the test passes while actually failing silently. `nock.isDone()` can be used to assert that all nocked endpoints were called (example in [On-Duty Bot](https://github.com/ampproject/amp-github-apps/blob/master/onduty/test/e2e.test.ts#L28))

Reply:
Good call, added.